### PR TITLE
Update activity contractor costs section

### LINKED ISSUE #1522: Implement the Private Contractor Costs subsection using CMS design system
From the design ticket #1463 we are ready to implement!

Final mockup:

### This task is done when...
 - [ ] The Private Contractor Costs section looks like the mockup
 - [ ] Use the [review pattern](https://design.cms.gov/patterns/review/) from the CMS design system  / the [Key Personnel pattern](https://hitech-apd.app.cloud.gov/apd#apd-state-profile-key-personnel) from the current app
 - [ ] Designers approve! 
 
Details:
- This design was decided upon on 5/7 as part of our post-standup parking lot session.
- Don't worry about the exact pixels heights from this mockup, I am using the [review pattern](https://design.cms.gov/patterns/review/) from the CMS design system.
- The action to edit opens the section up again, like [the Key Personnel section](https://hitech-apd.app.cloud.gov/apd#apd-state-profile-key-personnel).
- The action to remove prompts you with an "Are you sure?" modal.
- Bold the labels, this pattern does not exist in other parts of the CMS (like [the Key Personnel section](https://hitech-apd.app.cloud.gov/apd#apd-state-profile-key-personnel)), but it is a pattern the team would like to follow for the review section.
- Don't truncate text, leave her as long as she wants to be!

### PR DESCRIPTION
**Pull request by mgwalker**
### This pull request changes...
- updates the activity contractor costs section
- fixes the dev seed to have correct data
- closes #1522

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/seeds/development/apds.js [hunk 1]
```diff
@@ -407,11 +407,44 @@ exports.seed = async knex => {
...
+              { contractor_resource_id: g, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: g, hours: 0, rate: 0, year: 2020 },
+              { contractor_resource_id: h, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: h, hours: 0, rate: 0, year: 2020 },
+              { contractor_resource_id: i, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: i, hours: 0, rate: 0, year: 2020 },
+              { contractor_resource_id: j, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: j, hours: 0, rate: 0, year: 2020 },
+              { contractor_resource_id: k, hours: 0, rate: 0, year: 2019 },
+              { contractor_resource_id: k, hours: 0, rate: 0, year: 2020 }
```
Comment (mgwalker) [inline api/seeds/development/apds.js]: The hourly data is required. When an APD is created, the API inserts all this as initial data, but our seed was leaving it out. 😬

**Comment by jeromeleecms** [comment]
@mgwalker Not sure if this is related or not, but when I'm working in the sample APD and click on the first private contractor: 

image

The screen blanks out. Confirmed the behavior on IE and Firefox. Subsequent private contractors don't have this behavior and it opens like normal:

**Comment by jeromeleecms** [comment]
@mgwalker Also, when the hourly resource "Yes" selection is made, the "Done" button appear on the same line as the dollar value:

image

**Comment by mgwalker** [comment]
@jeromeleecms The first one is due to faulty data in the sample APD. 😬 I've updated your APD if you want to try it again.  I'll reset everyone else's as well.

**Comment by jeromeleecms** [comment]
Changes above look good, but encountering this:

image

Per Retro, posting here to determine if this is a bug or not before approval.

**Comment by mgwalker** [comment]
@jeromeleecms Is the save error still happening for you? I can't get it to happen, but if it's happening consistently for you, maybe we can pair up sometime to look into it?

**Comment by jeromeleecms** [comment]
@mgwalker still getting the error:

image

I'll ping you on Slack tomorrow to check your available to pair on this.

File: web/src/containers/activity/ContractorResourcesForm.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  const toggleCollapsed = useCallback(e => {
+    e.preventDefault();
+    setCollapsed(prev => !prev);
+  }, []);
+
+  const startDate = useMemo(() => dateSplitter(contractor.start), [
+    contractor.start
+  ]);
+
+  const endDate = useMemo(() => dateSplitter(contractor.end), [contractor.end]);
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/containers/activity/ContractorResourcesForm.js]: That is a sensible request, to which the answer is...  can I please not?  😆

`prettier` is doing this (based on line length, I think).  I can make the change manually in a different editor (I've got VS Code configured to automatically run prettier on every save), but it'll just revert the next time I poke round in this file. 😛 I've taken the attitude of not fighting with prettier anymore and just accept that it's weird at the edges.

File: web/src/containers/activity/ContractorResourcesForm.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+              ))}
+            </div>
+          }
+        >
+          Yes
+        </Choice>
+      </fieldset>
+
+      {contractor.hourly.useHourly ? (
+        <p className="ds-u-margin-bottom--0">
```
[the remark this answers is not in the document]
Comment (mgwalker) [reply web/src/containers/activity/ContractorResourcesForm.js]: I *think* this was fixed in 9efce0f where I wrapped the hourly boxes in the `<p>` tags. Prior to that, they were naked and inline. 😬

**Comment by mgwalker** [comment]
@jeromeleecms I can reproduce the save error. Looking into it!  I think it's related to the start/end dates...

**Comment by mgwalker** [comment]
@jeromeleecms Looks like it's fixed!  Thanks for finding that.  😄

**Comment by austinhernandez** [comment]
Screen Shot 2019-05-28 at 12 55 53 PM

 - [ ] use the spacing from key personnel for the summary section
 - [ ] system notification to let me know i'm about to delete an important thing. (It's kinda close to the edit button so I could see this bein' an accidental deletion.)

*Missing data design*
Screen Shot 2019-05-28 at 12 55 37 PM
I'd love to get @jameshupp 's thoughts on what the content should be for the empty fields in this section. Also, If it's a `null` response for date, we should get a `null` for all the missing and important fields.

 - [ ] final content for missing data in the input fields @jameshupp and @NAretakis @jeromeleecms 
 - [ ] add an `<emphasis>` style to the `dates not completed` / no value text
 - [ ] @jeromeleecms and @NAretakis is important to specify no value was put in these $ fields or is it ok to default to $0?

*Here is a screenshot of the labels from Key Personnel for cross-reference:*
Screen Shot 2019-05-28 at 12 57 49 PM
 - [ ] bold labels and add colons at the end of the label to match the contractor costs section
 - [ ] system notification to let me know i'm about to delete the section

**Comment by austinhernandez** [comment]
I was wrong about spacing. Please disregard the above about spacing and look at this mockup!
small top padding, larger bottom padding between the summary block (no label) and the labeled fields. 

image

**Comment by mgwalker** [comment]
- fixed the vertical spacing between the contractor name and description of services
- added a confirmation prompt before deleting the contractor
- added placeholder text for when the name and description are empty
- created #1558 to capture changes needed for the key personnel section

**Comment by jameshupp** [comment]
> I'd love to get @jameshupp 's thoughts on what the content should be for the empty fields in this section. Also, If it's a `null` response for date, we should get a `null` for all the missing and important fields.
> 
I think I need more info here. Are we treating these as errors, or are we just telling someone they're empty? Probably want to write them slightly differently depending on which.

**Pull request merged**